scripts/cpplint.py

```diff
@@ -3972,14 +3972,6 @@ def CheckBracesSpacing(filename, clean_lines, linenum, nesting_state, error):
       error(filename, linenum, 'whitespace/braces', 5,
             'Missing space before {')
 
-  # Make sure '} else {' has spaces.
-  # if Search(r'}else', line):
-  #   error(filename, linenum, 'whitespace/braces', 5,
-  #        'Missing space before else')
-  if (Search(r'^.*[^\s].*}$', line) or Search(r'^.*[^\s].*{$', line)) and not(Search(r'{[^}]*}', line)):
-     error(filename, linenum, 'whitespace/braces', 5,
-           'Put braces on a separate next line')
-
```

Comment:
Doesn't this also remove the check for all compound statements such as loops, non-lambda function bodies, if/else? How about, instead of completely removing it, adding a `not(Search(r'\]\s*\(.*\)\s*{', line))`?

Reply:
That doesn't work in the case of lambda with a return type:

```c++
[] (T& t) -> optionalt<std::string> { ...
```

Or with a lambda with omitted arguments (used in named scopes):

```c++
auto value = [&] {
  if (this)
    return;
  if (that)
    return;
  return;
}();
```

Reply:
It wouldn't be very hard to add expressions for those, but I do worry about the various line breaks that could be used.